### PR TITLE
React Swipeable Routes Added

### LINKED ISSUE #39: Create swipeable views to swipe to change between the pages.
## Adding Swipeable Views
Since the app is focused for mobile we can use in-built feature of material-ui to make the views swipeable . Swipeable in the sense that the user can navigate through the pages swiping left or right. 

## React-swipeable-views 
To be able to make the views swipeable .We need to drop React Routing from the views. If the team is not planning on lazy loading the pages or code splitting, this feature is a must since it will give the user a mobile like experience.A major refractor needs to be done to support swipeable views.
Refer:-https://material-ui.com/components/tabs/  
abc

### PR DESCRIPTION
**Pull request by ADRE9**
**Issue Number**

fixes #39 

**Changes Made**
1.I removed the Switch component as ReactSwipeable was doing the same thing either way.
2. I removed out the MainLayout component from every page since it was unnecessary and redundant.
3. I added the MainLayout over the ReactSwipeable component.
4. I had to use useEffect to update location as the user swipes so as the location reflects in the Bottom Navigation.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**
A Warning pops up in yellow as:-

Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
Please update the following components: ReactSwipableView

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
Make PR against master, and rebase to latest changes of master since it has linting errors being fixed.

**Comment by ADRE9** [comment]
I've made the PR to the main now. Can you please check if everything is right? @Abhishek-kumar09

File: package.json [hunk 1]
```diff
@@ -32,7 +32,12 @@
     "react-router": "^5.1.2",
     "react-router-dom": "^5.1.2",
     "react-router-hash-link": "^2.4.0",
+<<<<<<< HEAD
+    "react-scripts": "4.0.3",
+    "react-swipeable-routes": "^0.8.0",
+=======
     "react-scripts": "^4.0.3",
+>>>>>>> 73e315749cd4c01d6402edc9d6ace824c9ca9c87
```
Comment (Abhishek-kumar09) [inline package.json:40]: You have not resolved conflicts.
```diff
- <<<<<<< HEAD
+    "react-scripts": "4.0.3",
+    "react-swipeable-routes": "^0.8.0",
- =======
-    "react-scripts": "^4.0.3",
- >>>>>>> 73e315749cd4c01d6402edc9d6ace824c9ca9c87

```

File: src/Routes.js [hunk 2]
```diff
@@ -9,6 +10,39 @@ import WishlistPage from "./pages/wishListView";
 import ProfilePage from "./pages/ProfilePage";
 
 const renderRoutes = () => (
+<<<<<<< HEAD
```
Comment (Abhishek-kumar09) [inline src/Routes.js:13]: Conflicts Not resolved

File: src/layouts/MainLayout/BottomNav.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
+<<<<<<< HEAD
+import React,{useEffect} from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import BottomNavigation from '@material-ui/core/BottomNavigation';
+import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
+import {HomeOutlined, PlayCircleOutline, AccountCircleOutlined, AddCircleOutlineOutlined} from '@material-ui/icons';
+import { useHistory, useLocation } from 'react-router';
+=======
```
Comment (Abhishek-kumar09) [inline src/layouts/MainLayout/BottomNav.js:8]: same

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Conflicts are not resolved, see how to resolve it: https://www.youtube.com/watch?v=JtIX3HJKwfo

File: src/layouts/MainLayout/BottomNav.js (hunk 3 above)
Comment (ADRE9) [reply src/layouts/MainLayout/BottomNav.js]: Thanks for the video you shared. Well I am learning loads.🙂

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,13123 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (Abhishek-kumar09) [inline yarn.lock:1]: Remove yarn.lock I have added yar.lock in gitignore in latest code.It was hard to manage yarn.lock and package-lock.json at the same time.

Comment (ADRE9) [reply yarn.lock:1]: I am on it.🙌🏻

Comment (ADRE9) [reply yarn.lock]: Done @Abhishek-kumar09 .

File: src/layouts/MainLayout/BottomNav.js [hunk 5]
```diff
@@ -26,13 +26,27 @@ export default function BottomNav() {
...
+    if (location.pathname === '/home' && value !== "/home") {
+      setValue("/home");
+    }else if (location.pathname === '/personal' && value !== "/personal") {
+      setValue("/personal");
+    }else if (location.pathname === '/wishlist' && value !== "/wishlist") {
+      setValue("/wishlist");
+    }else if (location.pathname === '/profile' && value !== "/profile") {
+      setValue("/profile");
+    }
+  },[value,location]);
```
Comment (Abhishek-kumar09) [inline src/layouts/MainLayout/BottomNav.js]: This is not needed.
Any reason for which you added this?

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Don't forget to run npm install once you have added a new package in package.json.
And please verify that you don't have linting errors before pushinng.

**Comment by Abhishek-kumar09** [comment]
Thanks for your PR. Merging it.

**Pull request merged**